django-stubs/db/models/fields/json.pyi

```diff
@@ -1,34 +1,100 @@
-    def get_internal_type(self) -> Any: ...
-    def get_prep_value(self, value: Any) -> Any: ...
     def get_transform(self, name: Any) -> Any: ...
-    def validate(self, value: Any, model_instance: Any) -> None: ...
     def value_to_string(self, obj: Any) -> Any: ...
-    def formfield(self, **kwargs: Any) -> Any: ...
+    @overload
+    def __init__(
+        self: JSONField[_A],
+        encoder: json.JSONEncoder = ...,
+        decoder: json.JSONDecoder = ...,
+        verbose_name: Optional[str] = ...,
+        name: Optional[str] = ...,
+        primary_key: bool = ...,
+        max_length: Optional[int] = ...,
+        unique: bool = ...,
+        blank: bool = ...,
+        null: Literal[False] = ...,
+        db_index: bool = ...,
+        default: Optional[Union[_A, Callable[[], _A]]] = ...,
+        editable: bool = ...,
+        auto_created: bool = ...,
+        serialize: bool = ...,
+        unique_for_date: Optional[str] = ...,
+        unique_for_month: Optional[str] = ...,
+        unique_for_year: Optional[str] = ...,
+        choices: Iterable[
+            Union[Tuple[_A, str], Tuple[str, Iterable[Tuple[_A, str]]]]
+        ] = ...,
+        help_text: str = ...,
+        db_column: Optional[str] = ...,
+        db_tablespace: Optional[str] = ...,
+        validators: Iterable[_ValidatorCallable] = ...,
+        error_messages: Optional[_ErrorMessagesToOverride] = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: JSONField[Optional[_A]],
+        encoder: json.JSONEncoder = ...,
+        decoder: json.JSONDecoder = ...,
+        verbose_name: Optional[str] = ...,
+        name: Optional[str] = ...,
+        primary_key: bool = ...,
+        max_length: Optional[int] = ...,
+        unique: bool = ...,
+        blank: bool = ...,
+        null: Literal[True] = ...,
+        db_index: bool = ...,
+        default: Optional[Union[_A, Callable[[], _A]]] = ...,
+        editable: bool = ...,
+        auto_created: bool = ...,
+        serialize: bool = ...,
+        unique_for_date: Optional[str] = ...,
+        unique_for_month: Optional[str] = ...,
+        unique_for_year: Optional[str] = ...,
+        choices: Iterable[
+            Union[Tuple[_A, str], Tuple[str, Iterable[Tuple[_A, str]]]]
+        ] = ...,
+        help_text: str = ...,
+        db_column: Optional[str] = ...,
+        db_tablespace: Optional[str] = ...,
+        validators: Iterable[_ValidatorCallable] = ...,
+        error_messages: Optional[_ErrorMessagesToOverride] = ...,
+    ) -> None: ...
+    @overload
+    def __new__(  # type: ignore [misc]
+        cls,
+        *args: Any,
+        null: Literal[False] = ...,
+        choices: Iterable[
+            Union[Tuple[_A, str], Tuple[str, Iterable[Tuple[_A, str]]]]
+        ] = ...,
+        **kwargs: Any,
+    ) -> JSONField[_A]: ...
+    @overload
+    def __new__(
+        cls,
+        *args: Any,
```

Comment:
wondering if we can do better than `*args: Any` for `__new__` presumably there is a limit to what can be passed in

Reply:
This is just trying to simplificate things...

This is very similar to the [comment in here](https://github.com/sbdchd/django-types/pull/64/files/c60ed589c4cce2c947b3ca67c2db5f24768fb8a2#r739896552). In `__new__` we are just worried to make the `Literal[True/False]` magic happen and nothing else. Pyright and mypy will not use it to validate the fields passed to the input, so I simplified it by omitting everything else and just declaring what is important for the  `Literal[True/False]` magic, which are the null and choices fields.